### PR TITLE
feat: Add data commons toolkit

### LINKED ISSUE #935: [Feature Request] Integrate DataCommons
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

https://datacommons.org/

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dxmaptin**
## Description

Integrate DataCommons

## Motivation and Context

 `close #935 ` 

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/toolkits/data_commons.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+    - Analyzing property labels and values
+    - Retrieving places within a given place type
+    - Obtaining statistical values for specific variables and locations
+
+    All the data are grabbed from the knowledge graph of Data Commons.
+    Refer to https://datacommons.org/browser/ for more details.
+    """
+
+    @staticmethod
+    def query_data_commons(query_string, select_function=None):
```
Comment (Wendong-Fan) [inline camel/toolkits/data_commons.py:37]: missing type hint for argument and output

File: camel/toolkits/data_commons.py [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+        r"""Query the Data Commons knowledge graph using SPARQL.
+
+        Args:
+        query_string (str): A SPARQL query string.
+        select_function (callable, optional): A function to select rows to be
+            returned. It should take a dict as input and return a boolean.
+
+        Returns:
+        list: A list of dictionaries, each representing a node matching the
+        query conditions.
```
Comment (Wendong-Fan) [inline camel/toolkits/data_commons.py:47]: docstring format need to be aligned with above

File: camel/toolkits/data_commons.py [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+                    f"variable '{stat_var}'."
+                )
+                return {}
+            return result
+        except Exception as e:
+            print(f"An error occurred while querying Data Commons: {e!s}")
+            return {}
+
+    @staticmethod
+    def get_property_labels(dcids, out=True):
```
Comment (Wendong-Fan) [inline camel/toolkits/data_commons.py:204]: missing type hint

**Review by Wendong-Fan — COMMENTED** [review]
Thanks @dxmaptin 's contribution! docstring and type hint can be further polished, overall looks good to me

File: camel/toolkits/data_commons.py [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+            print(f"KeyError occurred: {e!s}")
+            print(
+                "One or more of the provided DCIDs do not exist in the "
+                "Data Commons knowledge graph."
+            )
+            return {}
+
+        except Exception as e:
+            print(f"An unexpected error occurred: {e!s}")
+            return {}
```
Comment (Wendong-Fan) [inline camel/toolkits/data_commons.py:149]: this is great to handle specific error, but from code concise and maintain's perspective, I would suggest just use 
```
        except Exception as e:
            return [{"error": f"An unexpected error occurred: {e!s}"}]
```
to catch all exceptions

File: camel/toolkits/data_commons.py [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+
+    @staticmethod
+    def get_stat_series(
```
Comment (Wendong-Fan) [inline camel/toolkits/data_commons.py:152]: rename to `get_stat_time_series`?

File: camel/toolkits/data_commons.py [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+                a certain format.
+            unit (str, optional): The unit of measurement.
+
+        Returns:
+            Dict[str, Any]: A dictionary containing the statistical time
+                series data.
+
+        Note:
+        - The function will raise a ValueError if any of the required
+          arguments are missing.
```
Comment (Wendong-Fan) [inline camel/toolkits/data_commons.py:180]: not necessary

File: camel/toolkits/data_commons.py [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+
+    @staticmethod
+    def get_property_labels(
+        dcids: Union[str, List[str]], out: bool = True
+    ) -> Dict[str, List[str]]:
+        r"""Retrieves and analyzes property labels for given DCIDs.
+
+        Args:
+            dcids (list): A list of Data Commons IDs (DCIDs) to analyze.
+            out (bool): Direction of properties to retrieve. Defaults to True.
```
Comment (Wendong-Fan) [inline camel/toolkits/data_commons.py:214]:
```suggestion
            out (bool): Whether or not the property points away from the given list of nodes. Defaults to True.
```

File: camel/toolkits/data_commons.py [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+            print(f"An error occurred while analyzing property labels: {e!s}")
+            return {}
+
+    @staticmethod
+    def get_property_values(
+        dcids,
+        prop,
+        out=True,
+        value_type=None,
+        limit=datacommons.utils._MAX_LIMIT,
```
Comment (Wendong-Fan) [inline camel/toolkits/data_commons.py:239]: pass the value directly after importing from datacommons

Comment (Wendong-Fan) [inline camel/toolkits/data_commons.py:239]: missing type hint, same as methods below

**Review by Wendong-Fan — COMMENTED** [review]
Thanks @dxmaptin ! Left some comments below

**Comment by dxmaptin** [comment]
Unit test passes and fixed other minor issues

**Pull request merged**